### PR TITLE
AIX_PB: create AdoptOpenJDK_AIX_Playbook/* - to replace current 'flatfile' aix.yml

### PR DESCRIPTION
**Pull request by aixtools**
A second start at developing a new playbook for AIX.

The structure is based on the AdoptOpenJDK_Unix_Playbook.

At the moment - this has been largely "cut/paste" from aix.yml to ./roles/{{ rolename }}/tasks/main.yml

Note: rebased to decrease noise.

### CODE REVIEW

**Comment by Haroon-Khel** [comment]
Though this is a duplicate of https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1434, I may close that one as some changes have been made to aix.yml, making it outdated

**Comment by aixtools** [comment]
Yes, I missed that - this is my learning curve.

FYI: I chose "AIX" as part of the directory name because:
```(py36) aixtools@x064:[/home/aixtools/ansible]ansible localhost -m setup | grep -i \"AIX\"
        "ansible_distribution": "AIX",
        "ansible_os_family": "AIX",
        "ansible_system": "AIX",
```

**Review by Haroon-Khel — CHANGES_REQUESTED** [review]
There needs to be a task which imports the variables in `adoptopenjdk_variables.yml`. You could encompass this task in a role which also includes the tasks which install bash

**Review by Haroon-Khel — CHANGES_REQUESTED** [review]
Despite yum's limitations, I think we should first focus on one thing at a time. That is, first replace the single aix playbook with a more structured one, and once thats finalised then we can focus on improving the yum role. 

Also, I think youve mistakenly included the `aixpm-play.yml` playbook into this pr? If it is only to be a playbook example, I dont think it should be pushed with the other commits

**Comment by aixtools** [comment]
> 
> 
> There needs to be a task which imports the variables in `adoptopenjdk_variables.yml`. You could encompass this task in a role which also includes the tasks which install bash

iirc - I was looking at making that a default action. iirc, having it in the `defaults/all` directory means it is processed automatically.

**Comment by aixtools** [comment]
> 
> 
> Despite yum's limitations, I think we should first focus on one thing at a time. That is, first replace the single aix playbook with a more structured one, and once thats finalised then we can focus on improving the yum role.
> 
> Also, I think youve mistakenly included the `aixpm-play.yml` playbook into this pr? If it is only to be a playbook example, I dont think it should be pushed with the other commits

I put it here - to assist with using the proposed role 'aixpm'.

And, while I agree it is not part of the final solution - is it part of the documentation - and I suppose, in that thought, it could be moved (in the end) to the documentation area of the role.

**Comment by aixtools** [comment]
I removed the new role 'aixpm' to simplify what is here 'right now'. My original goal is not changed, but the path getting there has - make this a simpler change - so it can be merged sooner - work on things such as re-writing the scripts (to no longer need them!) - will come later.

For the same reason - I'll also remove my 'AIX-Setup' document that describes a dual approach.

**Review by aixtools — COMMENTED** [review]
I do not see what 'github' actually expects of me here. I can only comment - but I do not see any change to my initial draft PR.

So, here is my comment - and let's see how github reacts :)

**Review by aixtools — COMMENTED** [review]
How about just renaming `adoptopenjdk_variables.yml` to` main.yml`?

**Review by aixtools — COMMENTED** [review]
I have removed both the playbook you mentioned - as well as the proposed role.

per private messages: 1st step is getting the existing playbook restructured. Changes to content come later.

**Review by aixtools — COMMENTED** [review]
Yet another 'review' block to fill in.

Removed proposed role named aixpm - for later consideration.

**Review by aixtools — COMMENTED** [review]
Do these ever close? The 'review requests? - Getting 'stuck' in the mechanisms.

**Comment by karianna** [comment]
@aixtools We're all tied up with the release of Java 8, 11, and 15 this week :-)

**Comment by aixtools** [comment]
Thought that was last week. I'll be patient!

* Thx for the heads up. Much appreciated!

**Comment by gdams** [comment]
I haven't done a full code review yet but so far this is looking great! I would like to reach a point where we could merge this into the main Unix playbook. The roles could be easily lifted across. 

Do we have a way to easily test these changes?

**Comment by Haroon-Khel** [comment]
Im in the process of testing it on an aix test environment. I got a bit distracted with other issues recently, but have now time to resume testing this

**Comment by aixtools** [comment]
> 
> 
> Im in the process of testing it on an aix test environment. I got a bit distracted with other issues recently, but have now time to resume testing this

Great!

**Review by Haroon-Khel — CHANGES_REQUESTED** [review]
I like the look of it. The tasks in the `tasks` block, in `main.yml`, still need to be added to their own roles as they currently run after the tasks listed under `roles`. Obviously, tasks like `name: Check if bash is installed` need to run first

**Comment by aixtools** [comment]
> 
> 
> I like the look of it. The tasks in the `tasks` block, in `main.yml`, still need to be added to their own roles as they currently run after the tasks listed under `roles`. Obviously, tasks like `name: Check if bash is installed` need to run first

* Not sure why having bash needs to run first - but this can be moved to a role - and run as the first role AFTER yum processing - so that the `yum:` module can work.

* If I look at the AdoptOpenJDK_Unix_Playbook/main.yml I see that the `tasks:` block appears earlier in the file. I am new to Ansible, but my understanding was that Ansible processed `tasks:` and `roles:` in a specific order, regardless of there order in the **YAML** code.

* I'll move the tasks block up to align with the other playbook - and shall also modify the PATH variable in the same way. There is nothing I have seen that indicates /opt/freeware/bin needs to be first in PATH via /etc/environment.

* I have forgotten the command - I mentioned it in a slack chat - but since GNU grep does not have the `-p` argument there is at least a second AIX systems maintenance command that fails with `/opt/freeware/bin/grep` first in the PATH.

**Comment by aixtools** [comment]
I took a lighter _touch_ only moving the installation of bash to the role `yum`.

Just thought of something - the scripts want to call bash - while ksh works fine (there are no bash specific issues).

And recall the roles/aixfs/tasks/main.yml WILL be re-written to use Ansible rather than a script - again negating any early requirement for bash.

**Comment by aixtools** [comment]
This is failing the macOS check - but I do not understand how this has anything to do with the AIX playbook changes here.

image

**Comment by gdams** [comment]
> This is failing the macOS check - but I do not understand how this has anything to do with the AIX playbook changes here.

@aixtools I'll investigate why this is failing. Please ignore the macOS test failure

**Comment by aixtools** [comment]
Thx. Was researching. I see that #1660 and #1585 are also failing. Surprised to see that #1539 doesn't have the same CI checks. Perhaps AIX PR should also have a different list (if possible - off topic, so feel free to ignore - other than tell me how to make a better request.)

And - renaming subject line to AIX_PB - to look more like the "UnixPB" subject lines.

**Comment by gdams** [comment]
@aixtools we currently run the macOS playbook on all changes to the ansible directory (https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/.github/workflows/build_mac.yml#L7). I'll work on improving this

**Comment by Haroon-Khel** [comment]
@aixtools Not sure why I mentioned that the bash tasks needs to be run first, I meant to say the `Yum` role. The Yum role includes the installation of `tar` and `unzip`, both of which are needed on the remote machine for the later tasks which unzip packages, else ansible throws an error

**Comment by aixtools** [comment]
ok. just following the order (of roles) as they are now in `aix.yml`. If the role `yum` is needed first, easy enough to move it to the front.

Are you absolutely that GNU tar is needed? Occasionally I run into situations where a special feature is used and GNU tar is needed - but in 98%+ of the cases "AIX tar" works fine.

For now - yum: for unzip, but I have another way for that - with much less overhead. For later.

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/group_vars/all/main.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+# Domain for setting hostname
+Domain: adoptopenjdk.net
+
+# Sudoers file
+Sudoers_File: /etc/sudoers
+
+# Jenkins User Variables:
+Jenkins_Username: jenkins
+
```
[the remark this answers is not in the document]
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/group_vars/all/main.yml:18]: Yes. I was wondering where those were coming from.

I'll add this - but where do Vendor_Files (i.e, the key files) come from?

Or is that something kept out of the Ansible workflow?

**Review by Haroon-Khel — APPROVED** [review]

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/group_vars/all/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/group_vars/all/main.yml:19]: its a directory saved on the machine of anyone who runs the playbook. Usually contains keys

**Pull request merged**